### PR TITLE
Add refinery max iterations arg

### PR DESCRIPTION
**Pull request by AroneyS**
Can be set to 0 to skip refinery (though it does still run CheckM)

### CODE REVIEW

**Comment by rhysnewell** [comment]
Looks good, though is the confirmation that refining gets skipped when the parameter is set to 0 just from personal tests? The refining script didn't need any changes for that to work?

**Comment by AroneyS** [comment]
> Looks good, though is the confirmation that refining gets skipped when the parameter is set to 0 just from personal tests? The refining script didn't need any changes for that to work?

Yeh, just from testing. The script uses a while loop that wont run if =0. Though if we want to skip refinery we should really skip the CheckM rules too...

**Review by rhysnewell — APPROVED** [review]
I would prefer if checkm remains to be run by default so users get both checkm1/2 results so happy with how this runs now, users could opt in to skip it at some point. But we'd also need to add in a way to calculate the MAG stats that checkm1 does, wouldn't be too hard but just extra busy work.

**Comment by AroneyS** [comment]
Oh definitely, I just meant the metabat2_checkm, semibin_checkm and rosella_checkm

**Comment by rhysnewell** [comment]
Ah gotcha, yeah they should get skipped for sure but this is good for now. Maybe a way to feed the iteration parameter into those checkm rules, if it's zero then just generate the expected file and exit. I believe there is already a similar check in the rule to make sure bins actually exist in the folders, maybe can use that

**Comment by AroneyS** [comment]
> Ah gotcha, yeah they should get skipped for sure but this is good for now. Maybe a way to feed the iteration parameter into those checkm rules, if it's zero then just generate the expected file and exit. I believe there is already a similar check in the rule to make sure bins actually exist in the folders, maybe can use that

Good idea. I'll add that here. Also need to add the bin check to Metabat2 and Semibin. I have had Metabat2 return no bins when other bins do.

**Comment by AroneyS** [comment]
I moved the logic to python script. Should skip checkm_metabat/semibin/rosella when refinery iterations is 0 or when there are no bins detected

File: aviary/modules/binning/scripts/run_checkm.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+def checkm(checkm2_db, bin_folder, bin_ext, refinery_max_iterations, output_folder, output_file, threads):
+    if len([f for f in os.listdir(bin_folder) if f.endswith(bin_ext)]) > 0 and refinery_max_iterations > 0:
+        print(f"Using CheckM2 database {checkm2_db}/uniref100.KO.1.dmnd")
+        subprocess.run(f"CHECKM2DB={checkm2_db}/uniref100.KO.1.dmnd checkm2 predict -i {bin_folder}/ -x {bin_ext} -o {output_folder} -t {threads} --force")
+        shutil.copy(f"{output_folder}/quality_report.tsv", output_file)
+    else:
+        print(f"No bins found in {bin_folder}")
+        os.makedirs(output_folder)
+        Path(output_file).touch()
```
Comment (rhysnewell) [inline aviary/modules/binning/scripts/run_checkm.py:14]: Maybe a less scary message should be printed when `refinery_max_iterations==0` so users don't get confused

**Review by rhysnewell — COMMENTED** [review]
Looks good, just add an alternative debug message to delineate between the case when no bins were found for a binner and max refinery iterations was set to 0

**Comment by rhysnewell** [comment]
All good now, ready to merge?

**Comment by AroneyS** [comment]
How do you sign commits?

**Comment by rhysnewell** [comment]
https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

My understanding was that you have to be using `ssh` rather than `https` when you push and pull from the repo, might not be possible with the qut system now that I think about it. I'll revert that requirement for pull requests, just means that the commits that you send through won't be verified by github and you won't get credit for them on your homepage

**Pull request merged**